.github/workflows/ci-pipeline.yml

```diff
@@ -0,0 +1,53 @@
+name: Continuous Integration
+
+on: 
+  push:
+    paths-ignore:
+      - "**.md"
+      - "documentation/**"
+  pull_request: 
+    paths-ignore:
+      - "**.md"
+      - "documentation/**"
+  schedule:
+    - cron: "0 22 * * 5" #At 10PM every Friday
+
+jobs:
+  build:
+    name: Provision the test container and run the tests
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout repo content 
+      uses: actions/checkout@v2
+
+    - name: Create .env file
+      env:
+        API_KEY: ${{ secrets.API_KEY }}
+        API_TOKEN: ${{ secrets.API_TOKEN }}
+        BOARD: ${{ secrets.BOARD }}
+        NOTSTARTED_LIST: ${{ secrets.NOTSTARTED_LIST }}
+        INPROGRESS_LIST: ${{ secrets.INPROGRESS_LIST }}
+        COMPLETED_LIST: ${{ secrets.COMPLETED_LIST }}
+      run: |
+        sh create_env_file.sh
```

Comment:
Do you need this? I would have thought your tests would run without real secrets